aggregator/programs/build.ts

```diff
@@ -5,103 +5,164 @@ import { dirname, parseArgs } from "../deps.ts";
+      );
+
+      const networkConfigHash = await shortContentHash(repoNetworkConfigPath);
+
+      networkConfigPaths = {
+        repo: repoNetworkConfigPath,
+        build: `networkConfig-${networkConfigHash}.json`,
+      };
+
+      // Need to replace this value with a build location because otherwise
+      // this file might not be included in the docker image
+      buildLine = `NETWORK_CONFIG_PATH=${networkConfigPaths.build}`;
+    }
+
+    buildDotEnvLines.push(buildLine);
   }
 
-  console.log("Processing", f);
-  await Deno.mkdir(dirname(`${buildDir}/ts/${f}`), { recursive: true });
-  await Deno.copyFile(f, `${buildDir}/ts/${f}`);
+  if (networkConfigPaths !== nil) {
+    await Deno.copyFile(
+      networkConfigPaths.repo,
+      `${buildDir}/${networkConfigPaths.build}`,
+    );
+  }
+
+  await Deno.writeTextFile(`${buildDir}/.env`, buildDotEnvLines.join("\n"));
 }
 
-const sudoDockerArg = args["sudo-docker"] === true ? ["sudo"] : [];
+async function copyTypescriptFiles() {
+  for (const f of await allFiles()) {
+    if (!f.endsWith(".ts")) {
+      continue;
+    }
+
+    console.log("Processing", f);
+    await Deno.mkdir(dirname(`${buildDir}/ts/${f}`), { recursive: true });
+    await Deno.copyFile(f, `${buildDir}/ts/${f}`);
+  }
+}
 
-await shell.run(
-  ...sudoDockerArg,
-  "docker",
-  "build",
-  repoDir,
-  "-t",
-  `aggregator:${buildName}`,
-);
+async function buildDockerImage() {
+  const buildName = await BuildName();
 
-const dockerImageName = `aggregator-${buildName}-docker-image`;
+  const sudoDockerArg = args["sudo-docker"] === true ? ["sudo"] : [];
 
-await shell.run(
-  ...sudoDockerArg,
-  "docker",
-  "save",
-  "--output",
-  `${repoDir}/build/${dockerImageName}.tar`,
-  `aggregator:${buildName}`,
-);
+  await shell.run(
+    ...sudoDockerArg,
+    "docker",
+    "build",
+    repoDir,
+    "-t",
+    `aggregator:${buildName}`,
+  );
 
-if (sudoDockerArg.length > 0) {
-  // chown to the current user
-  const username = await shell.Line("whoami");
+  const dockerImageName = `aggregator-${buildName}-docker-image`;
 
   await shell.run(
-    "sudo",
-    "chown",
-    username,
+    ...sudoDockerArg,
+    "docker",
+    "save",
+    "--output",
     `${repoDir}/build/${dockerImageName}.tar`,
+    `aggregator:${buildName}`,
   );
-}
 
-await shell.run(
-  "gzip",
-  `${repoDir}/build/${dockerImageName}.tar`,
-);
+  if (sudoDockerArg.length > 0) {
+    // chown to the current user
+    const username = await shell.Line("whoami");
 
-console.log("Aggregator build complete");
+    await shell.run(
+      "sudo",
+      "chown",
+      username,
+      `${repoDir}/build/${dockerImageName}.tar`,
+    );
+  }
 
-async function allFiles() {
-  return [
-    ...await shell.Lines("git", "ls-files"),
-    ...await shell.Lines(
-      "git",
-      "ls-files",
-      "--others",
-      "--exclude-standard",
-    ),
-  ];
+  await shell.run(
+    "gzip",
+    `${repoDir}/build/${dockerImageName}.tar`,
+  );
```

Comment:
Can consider pushing this to a container image registry in the future such as Docker Hub. https://github.com/thehubbleproject/hubble-contracts/blob/master/.github/workflows/docker.yml